lib/tasks/unlock_invite_statuses.rake

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# This task is run inside Heroku Scheduler at regular intervals.
+# See Heroku Scheduler docs at: https://devcenter.heroku.com/articles/scheduler
+
+# Unlocks `InviteStatus` and `GroupInviteStatus` records that have been locked (stuck) for more than an hour.
+# This task sets locked statuses to the "unaccepted" status.
+task unlock_invite_statuses: :environment do
+  task_stats = {}
+  [InviteStatus, GroupInviteStatus].each do |invite_status_model|
+    model_name = invite_status_model.to_s.underscore
+    task_stats[model_name] = {}
+    task_stats["total_#{model_name.pluralize}"] = 0
+    invite_status_model::LOCKED_STATUSES.each do |status, _|
+      task_stats[model_name][status] = 0
+    end
+
+    invite_status_model
+      .where(status: invite_status_model::LOCKED_STATUSES)
+      .find_in_batches(batch_size: 100) do |invite_statuses|
+        invite_statuses.each do |invite_status|
+          time_difference = Time.now.utc - invite_status.updated_at.utc
+          next unless time_difference > 1.hour
+          task_stats[model_name][invite_status.status] += 1
+          invite_status.unaccepted!
```

Comment:
Suggest reporting an exception to haystack here. This should not happen, and any time that we do have to do this cleanup is a bad user experience and should be reported as an exception.